src/coreimage.cs

```diff
@@ -5673,85 +5498,3365 @@ interface CIKeystoneCorrection {
 	[Mac (10,15)]
 	[BaseType (typeof (CIKeystoneCorrection))]
-	interface CIKeystoneCorrectionCombined {
+	interface CIKeystoneCorrectionCombined : ICIKeystoneCorrectionCombinedProtocol {
+
+		[CoreImageFilterProperty ("outputTransform")]
+		CGAffineTransform OutputTransform { get; }
 	}
 
 	[CoreImageFilter]
 	[iOS (13,0)]
 	[TV (13,0)]
 	[Mac (10,15)]
 	[BaseType (typeof (CIKeystoneCorrection))]
-	interface CIKeystoneCorrectionHorizontal {
+	interface CIKeystoneCorrectionHorizontal : ICIKeystoneCorrectionHorizontalProtocol {
+
+#if false // no documentation about the type
+		[CoreImageFilterProperty ("outputRotationFilter")]
+		NSObject OutputRotationFilter { get; }
+#endif
+
+		[CoreImageFilterProperty ("outputTransform")]
+		CGAffineTransform OutputTransform { get; }
 	}
 
 	[CoreImageFilter]
 	[iOS (13,0)]
 	[TV (13,0)]
 	[Mac (10,15)]
 	[BaseType (typeof (CIKeystoneCorrection))]
-	interface CIKeystoneCorrectionVertical {
+	interface CIKeystoneCorrectionVertical : ICIKeystoneCorrectionVerticalProtocol {
+
+		[CoreImageFilterProperty ("outputRotationFilter")]
+		CGAffineTransform OutputRotationFilter { get; }
+
+		[CoreImageFilterProperty ("outputTransform")]
+		CGAffineTransform OutputTransform { get; }
 	}
 
 	[CoreImageFilter]
 	[iOS (13,0)]
 	[TV (13,0)]
 	[Mac (10,15)]
 	[BaseType (typeof (CIFilter))]
-	interface CIPerspectiveRotate {
-
-		[CoreImageFilterProperty ("inputFocalLength")]
-		float FocalLength { get; set; }
-
-		[CoreImageFilterProperty ("inputRoll")]
-		float Roll { get; set; }
-
-		[CoreImageFilterProperty ("inputPitch")]
-		float Pitch { get; set; }
+	interface CIPerspectiveRotate : ICIPerspectiveRotateProtocol {
 
-		[CoreImageFilterProperty ("inputYaw")]
-		float Yaw { get; set; }
+		[CoreImageFilterProperty ("outputTransform")]
+		CGAffineTransform OutputTransform { get; }
 	}
 
 	[CoreImageFilter]
 	[iOS (13,0)]
 	[TV (13,0)]
 	[Mac (10,15)]
 	[BaseType (typeof (CIFilter))]
-	interface CIGaborGradients {
+	interface CIGaborGradients : ICIGaborGradientsProtocol {
 	}
 
 	[CoreImageFilter]
 	[iOS (13,0)]
 	[TV (13,0)]
 	[Mac (10,15)]
 	[BaseType (typeof (CIFilter))]
-	interface CIRoundedRectangleGenerator {
+	interface CIRoundedRectangleGenerator : ICIRoundedRectangleGeneratorProtocol {
 
+#if !XAMCORE_4_0
+		[Obsolete ("Use 'InputExtent' instead.")]
 		[CoreImageFilterProperty ("inputExtent")]
 		CIVector Extent { get; set; }
+#endif
+	}
 
-		[CoreImageFilterProperty ("inputRadius")]
-		float Radius { get; set; }
+#region Protocols
 
-		[CoreImageFilterProperty ("inputColor")]
-		CIColor Color { get; set; }
+	interface ICIFilterProtocol {}
+
+	[iOS (13,0)]
+	[TV (13,0)]
+	[Mac (10,15)]
+	[Protocol (Name = "CIFilter")]
+	// CIFilter already exists so we're using the Swift name
+	interface CIFilterProtocol {
+
+		[Abstract]
+		[NullAllowed, Export ("outputImage")]
+		CIImage OutputImage { get; }
+
+		[Static]
+		[NullAllowed, Export ("customAttributes")]
+		NSDictionary<NSString, NSObject> CustomAttributes { get; }
```

Comment:
Do we know anything about the attributes? Strong dict will be nicer if possible.

Reply:
no :(
https://developer.apple.com/documentation/coreimage/cifilter/3228047-customattributes?language=objc